### PR TITLE
interact2D small feature update

### PR DESCRIPTION
**Pull request by ddkohler**
## Changes

* figure windows now have non-generic names ("Figure1" -> "interact2D {data name}")
* sideplot projections show the data extremes (min and max), rather than the mean.
* normalization of 2D map and side plot limits are now use the same code references (the `Norm` class)
* refactored interact2D to remove some of the 🍝 (just some).  

## Checklist

- [x] added tests, if applicable
- [x] updated documentation, if applicable
- [x] updated CHANGELOG.md
- [x] tests pass

### CODE REVIEW

File: WrightTools/artists/_interact.py [hunk 1]
```diff
@@ -153,16 +167,6 @@ def gen_ticklabels(points, signed=None):
...
-    if signed:
-        norm = np.nanmax(np.abs(arr))
-    else:
-        norm = np.nanmax(arr)
-    if norm != 0 and ignore_zero:
-        arr /= norm
-    return arr
-
-
 def interact2D(
```
Comment (kameyer226) [inline WrightTools/artists/_interact.py:170]: where did norm go?

**Review by kameyer226 — COMMENTED** [review]
Looks nice.

File: WrightTools/artists/_interact.py (hunk 1 above)
Comment (ddkohler) [reply WrightTools/artists/_interact.py:170]: this function was replaced by `Norm.__call__`

File: WrightTools/artists/_interact.py [hunk 2]
```diff
@@ -103,34 +103,48 @@ def get_colormap(signed):
...
+            if not self.current_state.local:
+                norm = mpl.colors.Normalize(vmin=channel.null, vmax=np.nanmax(channel[:]))
+            else:
+                norm = mpl.colors.Normalize(vmin=channel.null)
+                norm.autoscale_None(
+                    np.ma.masked_invalid(self.current_state.dat[channel.natural_name][:])
+                )
+            if norm.vmax == norm.vmin:
+                norm.vmax += 1
+        self.norm = norm
```
Comment (kameyer226) [inline WrightTools/artists/_interact.py:137]: I see `self.norm` generated here but `__call__` calls `self.norm(data)` .   This works?   I know the tests work but am surprised.

Comment (ddkohler) [reply WrightTools/artists/_interact.py:137]: @kameyer226 feel free test the code however you see fit.  Note that interact2D doesn't really have integrated tests in the same way e.g. Data objects do.  Basically, tests will only throw an error if interact2D fails to execute (e.g. throws an error).  We don't have assertions that verify the content of the plots.  I test manually by playing with the interact plots that are generated.

I am hoping reviewers will take interact2D for a spin with their own applications (perhaps @rpm4 is interested? ) to make sure it behaves under a variety of conditions.

Comment (kameyer226) [reply WrightTools/artists/_interact.py:137]: ok I understand now

**Review by kameyer226 — APPROVED** [review]

**Pull request merged**